cmp/cmpopts/transform.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2017, The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE.md file.
+
+package cmpopts
+
+import (
+	"reflect"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/internal/value"
+)
+
+// DiscardMapZeros returns a Transformer option that discards all map entries
+// with a zero value. This is useful for comparing two maps in such a way
+// that entries with the zero value are ignored.
+//
+// DiscardMapZeros can be used in conjunction with EquateEmpty,
+// but cannot be used with SortMaps.
+func DiscardMapZeros() cmp.Option {
```

Comment:
"zeros" and "zeroes" are both correct spellings for the plural of "zero". Perhaps it would be better to avoid that ambiguity if we can.

How about `DiscardZeroElements` or `OmitZeroElements`? (That could also apply to slices, e.g. for a slice of pointers with a nil pointer meaning "skip".)